### ISSUE TITLE
Creative Commons shouldn't be used for code

### ISSUE THREAD

**Issue by dhensby** [Quadra-Digital/silverstripe-mailchimp-module#1]
See https://creativecommons.org/faq/#can-i-apply-a-creative-commons-license-to-software

Also, you're using a non-commercial use licence which means that any dev using this module can't do so if they are commercially gaining from it and also no client can use it on a site that is making them money.

I'd recommend you consider a permissive open source licence such as [BSD](https://opensource.org/licenses/BSD-3-Clause), [ISC](https://opensource.org/licenses/ISC) or [MIT](https://opensource.org/licenses/MIT) licence

**Comment by HARVS1789UK** [comment]
Hi @dhensby thanks for the feedback regarding the licensing, I was aware that Creative Commons licences are normally used for things like photography and artwork rather than code however, everything I had read suggested that although it was un-common there is no reason not to. I had not read the comments in the article you have linked to.

I spent a couple of hours researching the options for a license, the reason I went for the CC-BY-NC-4.0 was because I could not find any other common license type which would prevent someone else selling our module for a profit? I am yet to find anything concrete to define what 'commercial purposes' actually means as it seems to be a pretty ambiguous term.

I had previously considered the potential limitations that 'non-commercial' could impose on the modules usage (i.e. users not being able to use the module on a site which makes them money) and decided that as I could not find a licence which does exactly what we want, we will be better served by using a licence with stricter requirements (and not bothering to enforce some elements) than we would be by using a licence that allows for more freedom.

I have looked at the 3 licence types you mention above (and a few others) before and, unless I am mistaken, they would all allow one of Quadra's competitors to clone our repo and start selling our module to their clients? That is not something I would be happy with. 

Do you know of any licence types which allow for this as I am yet to find any?

**Comment by dhensby** [comment]
In simple terms, if you don't allow commercial use then you only allow for personal and "fair" use. Basically someone can't generate revenue or profit from your work. With things like images, text and other creative media this is pretty straight forward "are you using my image and selling copies on a canvas, that's not allowed". For software it means that no developer can use the module in a commercial project (ie: one they are billing a client for).

By all means, if your intent is to not allow people to make money from your module, then you'll want a non-commercial licence, but really that means you're not talking about open source software.

Here's a quick read: https://www.quora.com/What-is-the-best-free-for-non-commercial-use-license-to-use-for-an-open-source-project/answer/Phillip-Rhodes

> I have looked at the 3 licence types you mention above (and a few others) before and, unless I am mistaken, they would all allow one of Quadra's competitors to clone our repo and start selling our module to their clients? That is not something I would be happy with.

Then you don't want an open source licence. SilverStripe itself (and pretty much all the modules in the eco-system) is free for commercial use, if they weren't they wouldn't be of much use to most of the developers.

Of course, you're free to licence your software as you like and to prevent commercial use of your work, but either lots of people will be breaching your licence (if they don't read it) or won't be able to use it commercially without you licencing them a copy.

> we will be better served by using a licence with stricter requirements (and not bothering to enforce some elements)

You can't really have a licence and just "promise" not to enforce bits of it (it might as well not be in the licence if that's the case) because you'll always have recourse to claim losses under it.

In summary, it's not really possible to have an open source licence for your software that prevents commercial use, as then it's just proprietary software that you're sharing the source code of. The best you can really do is to licence it and state that all modifications must be shared back with the original copyright owner, but that won't stop competitors using your product.

**Comment by dhensby** [comment]
A small follow up, if you see https://opensource.org/osd-annotated heading 6, you'll see that to be "open source" you can't restrict the commercial use of your product.

So having restrictions on commercial software effectively renders this module proprietary.

I admit, the terminology is ambiguous; the source code is clearly clearly being shared but that doesn't mean it conforms to what is meant when people say a project or module is "open source".

**Comment by HARVS1789UK** [comment]
@dhensby Im definitely in agreement with you about the ambiguity, especially the use of the word 'free' in most descriptions of open source (and the whole "Free as in speech, not as in beer" analogy).

We have discussed things internally and (within the development and marketing team at least) I think we are happy to use a more permissive licence. I have scheduled a meeting with our directors to discuss as ultimately it is their decision not mine.

Ill let you know what the outcome is.

**Comment by HARVS1789UK** [comment]
@dhensby - I have discussed with our Directors who share my willingness to make this truly 'Open Source'. I have therefore replaced the CC-BY-NC-4.0 licence with a BSD 3-Clause license (in commit 2464eeab73fce254503c202d370cd484bc82a33e).

Thanks for the feedback.

**Comment by dhensby** [comment]
Awesome :)
